department-of-veterans-affairs/caseflow
# Make attorney assignment error actionable

Resolves #13110

### Description
Pass and display the actual error to the front end and make some of our error messages more actionable. Also rename factory?

### Acceptance Criteria
- [x] See error generated by the back end on the front end

### Testing Plan
1. Log in as bvaaabshire and go to http://localhost:3000/queue/3/assign
2. Find an ama case and create a judge decision review task for it
```ruby
appeal = Appeal.find_by_uuid("<uuid>")
JudgeDecisionReviewTask.create!(appeal: appeal, assigned_to: User.third, parent: appeal.root_task)
```
3. Check case and an additional ama case and assign them both to an attorney
4. Confirm the error suggests the user refreshes the page, both cases remain checked, attorney case counts do not change, and the total number of cases does not change.
5. Confirm the above after refresh (minus the error of course)

### User Facing Changes
 - [x] More descriptive errors

 BEFORE|AFTER
 ---|---
![Screen Shot 2020-04-29 at 4 39 20 PM](https://user-images.githubusercontent.com/45575454/80644672-054dae00-8a38-11ea-8d9d-0925eae1ea2a.png)|![Screen Shot 2020-05-05 at 11 28 36 AM](https://user-images.githubusercontent.com/45575454/81084232-99e65f00-8ec3-11ea-8b9f-57b12ad3750a.png)
